Tests/swift-sdk-swift-tests/RequestCreatorTests.swift

```diff
@@ -29,6 +29,59 @@ class RequestCreatorTests: XCTestCase {
         TestUtils.validateMatch(keyPath: KeyPath(JsonKey.endUnreadMessageCount), value: inboxSession.endUnreadMessageCount, inDictionary: body)
     }
     
+    func testGetInAppMessagesRequestFailure() {
+        let expectation1 = expectation(description: "result succeeded when it should have failed")
+        let auth = Auth(userId: nil, email: nil)
+        let requestCreator = RequestCreator(apiKey: apiKey, auth: auth)
+        
+        let failingRequest = requestCreator.createGetInAppMessagesRequest(1)
+        
+        do {
+            _ = try failingRequest.get()
+        } catch {
+            expectation1.fulfill()
+        }
+        
+        wait(for: [expectation1], timeout: testExpectationTimeoutForInverted)
+    }
+    
+    func testGetInAppMessagesRequest() {
+        let inAppMessageRequestCount: NSNumber = 42
+        
+        let request = createRequestCreator().createGetInAppMessagesRequest(inAppMessageRequestCount)
+        let urlRequest = convertToUrlRequest(request)
+        
+        TestUtils.validate(request: urlRequest, requestType: .get, apiEndPoint: .ITBL_ENDPOINT_API, path: .ITBL_PATH_GET_INAPP_MESSAGES)
+        
+        guard let header = urlRequest.allHTTPHeaderFields else {
+            XCTFail("no header")
+            return
+        }
+        
+        XCTAssertEqual(header[AnyHashable.ITBL_HEADER_SDK_PLATFORM], String.ITBL_PLATFORM_IOS)
+        XCTAssertEqual(header[AnyHashable.ITBL_HEADER_SDK_VERSION], IterableAPI.sdkVersion)
+        XCTAssertEqual(header[AnyHashable.ITBL_HEADER_API_KEY], apiKey)
+        
+        do {
```

Comment:
Not needed for +1 but consider replacing do..catch with guard and try?
Usually try/catch semantics are not that great for Swift. That is why we have a soft try? which returns nil when it fails.

Reply:
so the code would look like
`guard let success = try? request else {
XCTFail(...) 
return
}
`